packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -93,7 +78,21 @@ export class BlockFetcher extends Fetcher {
    * @return {Promise}
    */
   async store(blocks: Array<any>) {
-    await this.chain.putBlocks(blocks)
+    if (blocks.length === 0) {
+      return
+    }
+    await this.chain.open()
+    blocks = blocks.map((b: Block) =>
+      Block.fromValuesArray(b.raw(), { common: this.config.common })
+    )
+    await this.chain.blockchain.initPromise
```

Comment:
Note: internally, in `blockchain`, `initPromise` is always awaited on methods where this is necessary.